packages/wdio-jasmine-framework/src/reporter.js

```diff
@@ -87,10 +98,47 @@ export default class JasmineReporter {
             })
         }
 
+        // errors on beforeAll are swallowed by Jasmine and can only be accessed later on
+        let errorsBeforeAll
+        let errorsAfterAll
+        if (suite.failedExpectations && suite.failedExpectations.length) {
+            let errors = suite.failedExpectations
+
+            if (this.shouldCleanStack) {
+                errors = suite.failedExpectations.map(x => this.cleanStack(x))
+            }
+
+            errorsBeforeAll = errors.filter(error => error.stack.includes('UserContext.beforeAll'))
+            errorsAfterAll = errors.filter(error => error.stack.includes('UserContext.afterAll'))
+        }
+
+        const hookBeforeAll = Object.assign({}, suite)
+        hookBeforeAll.type = 'hook'
+        hookBeforeAll.parent = parentSuite
+        hookBeforeAll.description = '"before all" hook'
+        if (errorsBeforeAll) {
+            hookBeforeAll.errors = errorsBeforeAll
+            hookBeforeAll.error = errorsBeforeAll[0]
+        }
+        this.emit('hook:end', hookBeforeAll)
```

Comment:
is it possible that we don't have before/after All hooks at all?

Reply:
Yes. But that's part of the problem... I couldn't come up with a way to find out if there are tests in `beforeAll` or `afterAll` with Jasmine, as Jasmine keeps them separated before the run only, then they all get put in a queue and get run all together. At this point I couldn't find a way to differentiate them again.

Reply:
We can detect before after All hooks in the end and trigger start event just before stop.
At least in such case we wont add not existing hooks

Reply:
I think there must be a misunderstanding... We can only know for sure at the end of the suite if there was a before / after hook if there was a failure in one of them. Else if there was a `beforeAll` that passed, we cannot differentiate it from a regular spec anymore, except if we can guarantee the name of the `beforeAll` was not defined. However, assuming nobody would specify a name for `beforeAll` seemed a bit dangerous to me. That's why I decided to consistently use the hooks. We could also just send them out if there was a failure, but this would still not be an accurate representation...